src/model/resources/flexible_ccs/allamcycle_commit.jl

```diff
@@ -0,0 +1,213 @@
+            +min(1,max(allam_dict[y, "min_power"][i],allam_dict[y, "ramp_dn"][i]))*allam_dict[y, "cap_size"][i]*vSHUT_Allam[y,i,t])
+
+    ### Minimum and maximum power output constraints 
+    @constraints(EP, begin
+        # Minimum stable power generated per technology "y" at hour "t" > Min power
+        [y in ALLAM_CYCLE_LOX, i in 1:2, t=1:T], EP[:vOutput_AllamcycleLOX][y,i,t] >= allam_dict[y, "min_power"][i]*allam_dict[y, "cap_size"][i]*vCOMMIT_Allam[y,i,t]
+    # Maximum power generated per technology "y" at hour "t" < Max power
+        [y in ALLAM_CYCLE_LOX, i in 1:2, t=1:T], EP[:vOutput_AllamcycleLOX][y,i,t] <= allam_dict[y, "cap_size"][i]*vCOMMIT_Allam[y,i,t]
+    end)
+
+    ### Minimum up and down times 
+    for y in ALLAM_CYCLE_LOX
+        for i in 1:2
+            ## up time
+            Up_Time = Int(floor(allam_dict[y, "up_time"][i]))
+            Up_Time_HOURS = [] # Set of hours in the summation term of the maximum up time constraint for the first subperiod of each representative period
+            for s in START_SUBPERIODS
+                Up_Time_HOURS = union(Up_Time_HOURS, (s+1):(s+Up_Time-1))
+            end
+
+            @constraints(EP, begin
+                # cUpTimeInterior: Constraint looks back over last n hours, where n = allam_dict[y, "up_time"][i])
+                [t in setdiff(INTERIOR_SUBPERIODS,Up_Time_HOURS)], vCOMMIT_Allam[y,i,t] >= sum(vSTART_Allam[y,i,e] for e=(t-allam_dict[y, "up_time"][i]):t)
+
+                # cUpTimeWrap: If n is greater than the number of subperiods left in the period, constraint wraps around to first hour of time series
+                # cUpTimeWrap constraint equivalant to: sum(vSTART_Allam[y,e] for e=(t-((t%hours_per_subperiod)-1):t))+sum(vSTART_Allam[y,e] for e=(hours_per_subperiod_max-(allam_dict[y, "up_time"][i])-(t%hours_per_subperiod))):hours_per_subperiod_max)
+                [t in Up_Time_HOURS], vCOMMIT_Allam[y,i,t] >= sum(vSTART_Allam[y,i,e] for e=(t-((t%hours_per_subperiod)-1):t))+sum(vSTART_Allam[y,i,e] for e=((t+hours_per_subperiod-(t%hours_per_subperiod))-(allam_dict[y, "up_time"][i]-(t%hours_per_subperiod))):(t+hours_per_subperiod-(t%hours_per_subperiod)))
+
+                # cUpTimeStart:
+                # NOTE: Expression t+hours_per_subperiod-(t%hours_per_subperiod) is equivalant to "hours_per_subperiod_max"
+                [t in START_SUBPERIODS], vCOMMIT_Allam[y,i,t] >= vSTART_Allam[y,i,t]+sum(vSTART_Allam[y,i,e] for e=((t+hours_per_subperiod-1)-(allam_dict[y, "up_time"][i]-1)):(t+hours_per_subperiod-1))
+            end)
+
+            ## down time
+            Down_Time = Int(floor(allam_dict[y, "down_time"][i]))
+            Down_Time_HOURS = [] # Set of hours in the summation term of the maximum down time constraint for the first subperiod of each representative period
+            for s in START_SUBPERIODS
+                Down_Time_HOURS = union(Down_Time_HOURS, (s+1):(s+Down_Time-1))
+            end
```

Comment:
Here and elsewhere, consider using `hoursbefore` and `hoursafter` in order to simplify formulation of the constraints. It should be possible to write a single constraint over `[t in 1:T]`.